pkg/compare/compare_test.go

```diff
@@ -329,6 +333,21 @@ error code:2`),
 			outputFormat: Json,
 			checks:       defaultChecks,
 		},
+		{
+			name:   "Allow Undefined Extras",
+			mode:   []Mode{DefaultMode},
+			checks: defaultChecks,
+		},
+		{
+			name:   "Check No Overwrite",
```

Comment:
can you change the name of **all of the tests** to be informative - should include that they test the "Allow Undefined Extras"\Strategic Merge and also what is the exact test case, example:

Strategic Merge cluster CR not Overwrites reference 

so it will be easy to understand by the name of the test\dir what exactly is this testcase